### ISSUE TITLE
ci-kubernetes-e2e-gce-federation started constantly failing since build 4503 (last Friday night)

### ISSUE THREAD

**Issue by dchen1107** [kubernetes/kubernetes#47737]
This is the first build on which ci-kubernetes-e2e-gce-federation started constantly failing: 

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-federation/4503/

cc/ @kubernetes/sig-federation-misc cc/ @csbell 

cc/ @kubernetes/kubernetes-release-managers @marun

**Comment by dchen1107** [comment]
@csbell I assigned this to you. Please triage or delegate, thanks!

**Comment by krzyzacy** [comment]
@madhusudancs something was leaking? Let me know if you need any help from infra side.

**Comment by madhusudancs** [comment]
@krzyzacy Doesn't look like a leak. Still not sure what's going on. Doesn't look like an infra failure at this point.

**Comment by madhusudancs** [comment]
**EOD Update**

TL;DR No resolution yet. I am still debugging the issue.

So far what we have found is that the discovery client we use to discover the available RBAC API version on the cluster where we host the federation control plane doesn't return the expected results. Due to this, neither the right service account is created nor the RBAC rules installed to give federation controller manager access to secrets that store the credentials to access the federated clusters. I am still not sure what is causing the RBAC discovery call to fail. Still debugging.

**Comment by madhusudancs** [comment]
cc @perotinus @irfanurrehman @liggitt @csbell

**Comment by liggitt** [comment]
@madhusudancs do you have the captured discovery doc (`/apis`) from the server?

**Comment by dchen1107** [comment]
#47980 was merged and cherrypicked to v1.7. Federation test looks fine now. We are dealing kubectl API discovery caching issue in #47977. Close this bug.
